Backend/Remora.Discord.Gateway/Transport/WebSocketPayloadTransportService.cs

```diff
@@ -247,8 +247,25 @@ public async Task<Result<IPayload>> ReceivePayloadAsync(CancellationToken ct = d
 
             do
             {
-                result = await _receiveRetryPolicy.ExecuteAsync(c => _clientWebSocket.ReceiveAsync(buffer, c), ct);
-
+                result = await _receiveRetryPolicy.ExecuteAsync
+                (
+                    async c =>
+                    {
+                        WebSocketReceiveResult result1;
+                        try
+                        {
+                            result1 = await _clientWebSocket.ReceiveAsync(buffer, c);
+                        }
+                        catch
+                        {
+                            await DisconnectAsync(true, c);
+                            throw;
```

Comment:
Disconnecting and re-throwing is not what should be done here; re-init the socket and attempt to connect again.

Reply:
ConnectAsync requires the Uri endpoint which idk if it can be obtained from the old socket (it might be possible though).

Reply:
I found a way to cache the endpoint by having ConnectAsync store it in an private field for this spot to use it.